src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -717,6 +728,14 @@ public void clearSqlWarnings() {
     sqlWarnings.clear();
   }
 
+  public SFConnectionHandler getSfConnectionHandler() {
+    return sfConnectionHandler;
```

Comment:
Note: needed so `BindUploader` can get a reference back to the connection handler from the `Session` contained within it. This is needed to resolve that connection handler's bind-stage name.